### ISSUE TITLE
Android ModelViewer Crash On Quick Creation and Detach

### ISSUE THREAD

**Issue by JustinBis** [google/filament#6933]
**Describe the bug**
Filament's `ModelViewer` helper for Android will crash if you detach the connected view quickly after creation.

I narrowed this down to the engine destruction within the detach handler used in ModelViewer:

```
// In filament-android's ModelViewer.kt
    private fun addDetachListener(view: android.view.View) {
        view.addOnAttachStateChangeListener(object : android.view.View.OnAttachStateChangeListener {
            override fun onViewAttachedToWindow(v: android.view.View) {}
            override fun onViewDetachedFromWindow(v: android.view.View) {
                uiHelper.detach()
                ...
                // This is the line that ultimately causes the crash
                engine.destroy()
            }
        })
    }
```

**To Reproduce**
1. Initialize a ModelViewer with a SurfaceView in some Fragment that you can quickly destroy, such as a bottom sheet
2. Quickly destroy the Fragment (e.g. dismiss the bottom sheet)
3. Observe the crash below

**Expected behavior**
Creating and immediately removing the view associated with a ModelViewer should not crash.

**Logs**
```
2023-07-01 13:15:12.526 25027-25027 libc                    ai.nokto.wire                        A  Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 25027 (ai.nokto.wire), pid 25027 (ai.nokto.wire)
2023-07-01 13:15:12.527  1018-24974 ExynosVide...Enc-Vp8Enc sam...hardware.media.c2@1.0-service  E  [updateAverageQpInfoToParam] Get_AverageQp() is failed
2023-07-01 13:15:12.545  1018-24974 ExynosVide...Enc-Vp8Enc sam...hardware.media.c2@1.0-service  E  [updateAverageQpInfoToParam] Get_AverageQp() is failed
---------------------------- PROCESS STARTED (25123) for package ai.nokto.wire ----------------------------
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A  *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A  Build fingerprint: 'google/bluejay/bluejay:13/TQ3A.230605.010/10121037:user/release-keys'
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A  Revision: 'MP1.0'
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A  ABI: 'arm64'
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A  Timestamp: 2023-07-01 13:15:12.634041120-0600
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A  Process uptime: 9s
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A  Cmdline: ai.nokto.wire
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A  pid: 25027, tid: 25027, name: ai.nokto.wire  >>> ai.nokto.wire <<<
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A  uid: 10274
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A  tagged_addr_ctrl: 0000000000000001 (PR_TAGGED_ADDR_ENABLE)
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A  signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A      x0  0000000000000000  x1  00000000000061c3  x2  0000000000000006  x3  0000007fc7d4e0e0
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A      x4  6f73736120656854  x5  6f73736120656854  x6  6f73736120656854  x7  7020646574616963
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A      x8  00000000000000f0  x9  0000007c3d2899e0  x10 0000000000000001  x11 0000007c3d2cb370
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A      x12 000000000000ead3  x13 746169636f737361  x14 0000000001848592  x15 000002e363723c9c
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A      x16 0000007c3d338d50  x17 0000007c3d313eb0  x18 0000007c5330e000  x19 00000000000061c3
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A      x20 00000000000061c3  x21 00000000ffffffff  x22 b400007b0ce4ca78  x23 b400007aed049090
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A      x24 b400007aed049220  x25 b4000079acdcd950  x26 0000007fc7d4e4ac  x27 0000007fc7d4e4a0
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A      x28 0000007fc7d4e3a0  x29 0000007fc7d4e160
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A      lr  0000007c3d2bb1c8  sp  0000007fc7d4e0c0  pc  0000007c3d2bb1f4  pst 0000000000001000
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A  backtrace:
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A        #00 pc 00000000000531f4  /apex/com.android.runtime/lib64/bionic/libc.so (abort+164) (BuildId: dc4001c2ef2dfc23467040797a96840c)
2023-07-01 13:15:12.834 25121-25121 DEBUG                   crash_dump64                         A        #01 pc 000000000013dad8  /data/app/~~rplkVPmz7VbLEgpq_1Ne9g==/ai.nokto.wire-Jhrjqq6VqGhid7LMFIgW8Q==/base.apk!libfilament-jni.so (offset 0xd70000) (BuildId: 5b3cb03609ecae0f2ead20fd6421ecf4300392ef)
```

**Smartphone:**
 - Device: Pixel 6a
 - OS: Android 13 (Build TQ3A.230605.010)

**Additional context**

This seems related to https://github.com/google/filament/issues/4724

I'm working around this by creating my own fork of ModelViewer and delaying engine destruction to some time in the future (arbitrarily 3 seconds after the view is detached). That's aligned with the suggestion from @romainguy in [the first comment on that issue](https://github.com/google/filament/issues/4724#issuecomment-940426066).

Ideally, immediate engine destruction shouldn't crash. Separate from a fix for that, the ModelViewer class should also fully allow the caller to control the lifecycle of the engine instead of trying to destroy it itself. Right now, you can pass in an engine but you effectively transfer lifecycle control of it to the ModelViewer after construction.

**Comment by romainguy** [comment]
Tracked by issue #4724

**Comment by JustinBis** [comment]
@romainguy #4724 is 2+ years old and larger scope than is really needed to fix this issue. The key thing to fix here is that Filament should not crash if you quickly create and destroy an engine. Fixing this issue is a necessary but not sufficient condition to fix #4724. Fixing this would also let Android developers implement patches to fix #4724 on their own.

**Comment by pixelflinger** [comment]
I can easily reproduce. Basically launch gltf_viewer and immediately press "back".

**Comment by JustinBis** [comment]
I'm currently shipping an experiment to roll Filament back to 1.36.0 in our app based on the comment in issue #7024  [saying the crash is was introduced in version 1.38.0](https://github.com/google/filament/issues/7024#issuecomment-1662626532). Will update with results when I have them, if the rollback fixes the crash.

**Comment by pixelflinger** [comment]
@JustinBis I'm confident I'll have a fix by tomorrow.
